core/src/main/scala/io/snappydata/SnappyStoreTableSizeProviders.scala

```diff
@@ -0,0 +1,194 @@
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata
+
+import java.sql.Connection
+import java.util.ArrayList
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.language.implicitConversions
+import scala.reflect.ClassTag
+
+import com.gemstone.gemfire.cache.execute.FunctionService
+import com.gemstone.gemfire.i18n.LogWriterI18n
+import com.gemstone.gemfire.internal.SystemTimer
+import com.pivotal.gemfirexd.internal.engine.Misc
+import com.pivotal.gemfirexd.internal.engine.distributed.{RegionSizeCalculatorFunction, GfxdListResultCollector, GfxdMessage}
+import io.snappydata.Constant._
+
+import org.apache.spark.sql.SnappyContext
+import org.apache.spark.sql.execution.ConnectionPool
+import org.apache.spark.sql.execution.columnar.ExternalStoreUtils
+import org.apache.spark.sql.execution.columnar.impl.ColumnFormatRelation
+import org.apache.spark.{Logging, SparkContext}
+import java.util.HashMap
+
+object StoreTableValueSizeProviderService extends Logging {
+  @volatile
+  private var tableSizeInfo = Map[String, Long]()
+  private var timer: SystemTimer = null
+
+
+  def start(sc: SparkContext): Unit = {
+    val delay =
+      sc.getConf.getOption("spark.snappy.calcTableSizeInterval")
+          .getOrElse(DEFAULT_CALC_TABLE_SIZE_SERVICE_INTERVAL).toString.toLong
+    timer = new SystemTimer(Misc.getGemFireCache.getDistributedSystem,
+      true, Misc.getGemFireCache.getLoggerI18n)
+    timer.schedule(calculateTableSizeTask(sc), delay, delay)
+  }
+
+  def stop: Unit = {
+    if (timer != null) {
+      timer.cancel()
+    }
+  }
+
+  def calculateTableSizeTask(sc: SparkContext): SystemTimer.SystemTimerTask = {
+    new SystemTimer.SystemTimerTask {
+      override def run2(): Unit = {
+        val currentTableSizeInfo = mutable.HashMap[String, Long]()
+
+        val result =
+          FunctionService.onMembers(GfxdMessage.getAllGfxdServers())
+              .withCollector(new GfxdListResultCollector())
+              .execute(RegionSizeCalculatorFunction.ID).getResult()
+
+        result.asInstanceOf[ArrayList[HashMap[String, Long]]]
+            .asScala.foreach(_.asScala.foreach(row => {
+          var totalSize: Long = row._2
+          if (currentTableSizeInfo.contains(row._1)) {
+            totalSize = currentTableSizeInfo.get(row._1).get + row._2
+          }
+          currentTableSizeInfo.put(row._1, totalSize)
+        }))
+
+          tableSizeInfo = currentTableSizeInfo.toMap
```

Comment:
Had added `Utils.immutableMap(currentTableSizeInfo)` to avoid copying like toMap does for better performance (though that slight perf difference probably won't matter here).